### DISCUSSION TITLE
Access Wavelog through Cloudflare ZTNA

### DISCUSSION THREAD

**Discussion by PretorianJ** [wavelog/wavelog#1333] [General]
I have cloudflare and Zero trust. 

I publish Applications that I host internally on my Virtualization(Unraid) and through ZTNA I can access these apps from anywhere with internet access as soon as I login through my cloudflare. 

Issue is every app I have is working except wavelog. 

Has anyone tried it and it worked? 

I am a Senior IT Engineer so I am fairly skillful doing this.  I just can't figure why wavelog does not work and every app I have it does. 

Thank you in advance,

**Comment by HB9HIL** [comment]
Hello @PretorianJ 

Difficult to say. In the end Wavelog works like any other LAMP Stack. There is just one thing we experienced in the past:
If you run Wavelog in a subdirectory with a super special Alias Root Directory or something like that it may not work. 
Do you have any error messages or what did you find out until now?

Cheers HB9HIL

**Comment by 5Yyhu** [comment]
By now, I'm sure @PretorianJ has probably solved his Cloudflare Tunnels question, but I wanted to post a comment with some details about some configuration items I had to change to help Cloudflare Tunnels work with Wavelog for the benefit of others who may be trying to set up a tunnel for remote access.

There's nothing special about setting up the Cloudflare side of the equation. There are a myriad of [guides](https://medium.com/design-bootcamp/how-to-setup-a-cloudflare-tunnel-and-expose-your-local-service-or-application-497f9cead2d3) that help outline the steps.

The trouble I ran into exists on the server side. I operate Wavelog in a docker container. In the `/var/www/html/application/config/docker/config.php` file within the container the hostname of the container is captured in:  `$config['base_url']	= 'https://192.168.1.XXX:8086/';` The IP address here is a local network address and not a rotable FQDN or IP address from the public internet. When configured this way, Cloudflare will route the traffic through the tunnel, but then Wavelog will provided the local network IP address as the "correct" destination. As such, if you are on the local network, you'll see the URL in the browser bar as 192.168.1.XXX, not the domain name attached to the Cloudflare tunnel. If you're running Wavelog in a VPS or you're not on the local network, the traffic will not route. You have to change the `config.php` so the  domain name replaces the local network IP address: `$config['base_url'] = 'https://wavelog.domain.com/';`. Save the `config.php` file after updating and your Cloudflare tunnel should route appropriately.
